cloudigrade/account/management/commands/add_account.py

```diff
@@ -24,17 +28,52 @@ def add_arguments(self, parser):
         """Require an ARN to be provided."""
         parser.add_argument('arn', help=_('Granted Role ARN'))
 
+    @transaction.atomic
```

Comment:
Do we want atomic to be our default behavior?

Reply:
Generally, I would say yes, _but_ I'm wary of leaving open long-running transactions due to the unknown duration of external API calls. That's why I shuffled the code to do all the API calls first and _then_ do the various saves.

If we want to push the lever to do atomic everywhere, I would be fine with that as long as we're careful like this about how we do API calls separated from write operations.

Reply:
I can easily go either way. Let's do what you have and if it looks like every time we're using the atomic decorator we can assess down the line.